### ISSUE TITLE
factorize should be more limited

### ISSUE THREAD

**Issue by sorawee** [racket/math#32]
It would be ideal if `factorize` is injective. It is not right now due to how `(factorize 0)` = `(factorize 1)` = `'()`. Changing its type to `Positive-Integer` is one possibility to accomplish this.

**Comment by sorawee** [comment]
Also, regardless of this, the type of the co-domain could be strengthen to `(Listof (List Positive-Integer Positive-Integer))`.

**Comment by soegaard** [comment]
Wait. Isn't `(factorize 0) = '()`  simply an mistake?

The empty product is 1, so an empty list of factors represents 1.

**Comment by pavpanchekha** [comment]
I agree, `(factorize 0)` should raise an error or return `((0 1))` or something else but definitely not return `'()`. The easiest fix is definitely to change the domain to `Positive-Integer`.
